src/main/java/com/limechain/trie/MemoryTrieAccessor.java

```diff
@@ -0,0 +1,154 @@
+            .map(NodeHandle::getUserData)
+            .map(NodeData::getValue);
+    }
+
+    @Override
+    public DeleteByPrefixResult deleteMultipleNodesByPrefix(Nibbles prefix, Long limit) {
+        Optional<NodeHandle<NodeData>> optionalNodeHandle = initialTrie.existingNode(prefix);
+
+        if (optionalNodeHandle.isEmpty()) {
+            return new DeleteByPrefixResult(0, true);
+        }
+
+        NodeHandle<NodeData> nodeHandle = optionalNodeHandle.get();
+        AtomicInteger deleted = new AtomicInteger(0);
+
+        for (Nibble nibble : Nibbles.ALL) {
+            nodeHandle.getChild(nibble)
+                .map(NodeHandle::getNodeIndex)
+                .ifPresent(childIndex -> initialTrie.deleteNodesRecursively(childIndex, limit, deleted));
+        }
+
+        if (limit != null && deleted.get() >= limit) {
+            return new DeleteByPrefixResult(deleted.get(), false);
+        }
+
+        if (nodeHandle.hasStorageValue()) {
+            initialTrie.deleteStorageNodeAt(nodeHandle.getFullKey());
+            deleted.incrementAndGet();
+        } else {
+            initialTrie.deleteInternalNodeAt(nodeHandle.getFullKey());
+        }
+
+        return new DeleteByPrefixResult(deleted.get(), true);
+    }
+
+    @Override
+    public Optional<Nibbles> getNextKey(Nibbles key) {
+        NodeHandle<NodeData> rootHandle = initialTrie.getRootNode().orElse(null);
+        return findNextKey(rootHandle, key, Nibbles.EMPTY);
+    }
+
+    private Optional<Nibbles> findNextKey(NodeHandle<NodeData> node, Nibbles prefix, Nibbles currentPath) {
+        if (node == null) {
+            return Optional.empty();
+        }
+
+        // If the current node is a leaf and the fullPath is greater than the prefix, it's a candidate.
+        if (node.getUserData() != null && node.getUserData().getValue() != null && currentPath.compareTo(prefix) > 0) {
+            return Optional.of(currentPath);
+        }
+
+        for (Nibble nibble : Nibbles.ALL) {
+            NodeHandle<NodeData> childNode = node.getChild(nibble).orElse(null);
+            if (childNode == null) continue;
+            Nibbles nextPath = currentPath.add(nibble).addAll(childNode.getPartialKey());
+            Optional<Nibbles> result = findNextKey(childNode, prefix, nextPath);
+            if (result.isPresent()) {
+                return result;
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public void persistChanges() {
+        for (ChildTrieAccessor value : loadedChildTries.values()) value.persistChanges();
+        loadedChildTries.clear();
+
+        trieStorage.updateTrieStorage(initialTrie, updates, StateVersion.V0);
+    }
+
+    /**
+     * Retrieves the Merkle root hash of the trie with the specified state version.
+     *
+     * @param version The state version.
+     * @return The Merkle root hash.
+     */
+    public byte[] getMerkleRoot(StateVersion version) {
```

Comment:
This method is not a part of the interface since we'll have differences in the implementation logic which would change the method signature. 
In the memory trie we create a separate trie accessor for each child trie and then get the root node from there, whereas in the disk one I plan to directly retrieve the merkle root from the node in the main trie which holds the referencing merkle value as storage.